pkg/cmd/cli/cmd/set/env.go

```diff
@@ -101,8 +99,8 @@ func NewCmdEnv(fullName string, f *clientcmd.Factory, in io.Reader, out, errout
 	cmd.Flags().StringSliceVarP(&filenames, "filename", "f", filenames, "Filename, directory, or URL to file to use to edit the resource.")
 	cmd.Flags().Bool("overwrite", true, "If true, allow environment to be overwritten, otherwise reject updates that overwrite existing environment.")
 	cmd.Flags().String("resource-version", "", "If non-empty, the labels update will only succeed if this is the current resource-version for the object. Only valid when specifying a single resource.")
-	cmd.Flags().StringP("output", "o", "", "Display the changed objects instead of updating them. One of: json|yaml.")
-	cmd.Flags().String("output-version", "", "Output the changed objects with the given version (default api-version).")
+
+	kcmdutil.AddPrinterFlags(cmd)
```

Comment:
@fabianofranz 
> How does the new factory method behave with --sort-by

I have tested it on almost all sub-commands (with `json`, `yaml`, `custom-columns`, `wide` output formats) so far and everything prints without issue. I have not seen if it actually sorts multiple output values as all sub-commands seem to only deal with a single resource at a time